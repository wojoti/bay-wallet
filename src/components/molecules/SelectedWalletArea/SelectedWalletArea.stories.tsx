import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {SelectedWalletArea} from 'components/molecules';

export default {
  title: 'Molecules/SelectedWalletArea',
  component: SelectedWalletArea,
  argTypes: {
    onChangePress: {action: 'onChangePress'},
  },
} as ComponentMeta<typeof SelectedWalletArea>;
const Template: ComponentStory<typeof SelectedWalletArea> = args => (
  <SelectedWalletArea {...args} />
);

export const Example = Template.bind({});
Example.args = {
  walletValue: '$ 1.000',
  walletName: 'Example Wallet',
  testId: '',
};
