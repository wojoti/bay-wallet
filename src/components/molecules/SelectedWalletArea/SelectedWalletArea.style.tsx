import {StyleSheet} from 'react-native';
import {secondary} from 'theme/colors';

const style = StyleSheet.create({
  wrapper: {
    width: '85%',
    marginBottom: 30,
  },
  button: {
    borderWidth: 0,
    height: 'auto',
  },
  buttonFont: {
    color: secondary,
    fontSize: 15,
    fontWeight: 'normal',
  },
});
export default style;
