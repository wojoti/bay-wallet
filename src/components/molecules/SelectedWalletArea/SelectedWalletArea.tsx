import {Button, Container, Header, Label} from 'components/atoms';
import {PropsWithChildren} from 'react';
import {iconGrey, pink} from 'theme/colors';
import style from './SelectedWalletArea.style';

export type SelectedWalletAreaProps = PropsWithChildren<{
  onChangePress: () => void;
  walletValue: string;
  walletName: string;
  testId?: string;
}>;

const SelectedWalletArea = ({
  walletValue,
  walletName,
  onChangePress,
  testId,
}: SelectedWalletAreaProps) => {
  return (
    <Container
      customStyle={style.wrapper}
      testId={testId}
      direction="row"
      justify="space-around">
      <Container flex={1} justify="flex-start">
        <Header
          spacing={1}
          size={28}
          color={pink}
          testId="selectedwalletarea-header">
          {walletValue}
        </Header>
        <Label size={15} color={iconGrey} testId="selectedwalletarea-label">
          {walletName}
        </Label>
      </Container>
      <Container flex={1} align="flex-end" justify="flex-end">
        <Button
          customStyle={style.button}
          customFontStyle={style.buttonFont}
          onPress={onChangePress}
          testId="selectedwalletarea-button">
          Change
        </Button>
      </Container>
    </Container>
  );
};

export default SelectedWalletArea;
