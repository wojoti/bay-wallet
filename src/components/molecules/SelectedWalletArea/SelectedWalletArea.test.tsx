import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import SelectedWalletArea, {
  SelectedWalletAreaProps,
} from './SelectedWalletArea';

const mockOnChangePress = jest.fn();
const props: SelectedWalletAreaProps = {
  testId: 'test-selectedwalletarea-id',
  walletValue: '$ 1.000',
  walletName: 'Example Wallet',
  onChangePress: mockOnChangePress,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<SelectedWalletArea {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render SelectedWalletArea', () => {
  render(<SelectedWalletArea {...props} />);
  const SelectedWalletAreaElement = screen.getByTestId(
    'test-selectedwalletarea-id',
  );
  expect(SelectedWalletAreaElement).toBeOnTheScreen();
});

test('should render SelectedWalletArea - header with value', () => {
  render(<SelectedWalletArea {...props} />);
  const SelectedWalletAreaElement = screen.getByTestId(
    'test-selectedwalletarea-id',
  );
  expect(SelectedWalletAreaElement).toBeOnTheScreen();
  const headerElement = within(SelectedWalletAreaElement).getByTestId(
    'selectedwalletarea-header',
  );
  expect(headerElement).toBeOnTheScreen();
  expect(headerElement).toHaveTextContent('$ 1.000');
});

test('should render SelectedWalletArea - label with name', () => {
  render(<SelectedWalletArea {...props} />);
  const SelectedWalletAreaElement = screen.getByTestId(
    'test-selectedwalletarea-id',
  );
  expect(SelectedWalletAreaElement).toBeOnTheScreen();
  const labelElement = within(SelectedWalletAreaElement).getByTestId(
    'selectedwalletarea-label',
  );
  expect(labelElement).toBeOnTheScreen();
  expect(labelElement).toHaveTextContent('Example Wallet');
});

test('should render SelectedWalletArea - button', () => {
  render(<SelectedWalletArea {...props} />);
  const SelectedWalletAreaElement = screen.getByTestId(
    'test-selectedwalletarea-id',
  );
  expect(SelectedWalletAreaElement).toBeOnTheScreen();
  const buttonElement = within(SelectedWalletAreaElement).getByTestId(
    'selectedwalletarea-button',
  );
  expect(buttonElement).toBeOnTheScreen();
});

test('should render SelectedWalletArea - handle onChangePress', () => {
  render(<SelectedWalletArea {...props} />);
  const SelectedWalletAreaElement = screen.getByTestId(
    'test-selectedwalletarea-id',
  );
  expect(SelectedWalletAreaElement).toBeOnTheScreen();
  const buttonElement = within(SelectedWalletAreaElement).getByTestId(
    'selectedwalletarea-button',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(mockOnChangePress).toBeCalledTimes(0);
  fireEvent.press(buttonElement);
  expect(mockOnChangePress).toBeCalledTimes(1);
});
