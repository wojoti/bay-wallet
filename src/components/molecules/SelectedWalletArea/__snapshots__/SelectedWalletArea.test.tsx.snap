// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": undefined,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "width": "100%",
      },
      {
        "marginBottom": 30,
        "width": "85%",
      },
    ]
  }
  testID="test-selectedwalletarea-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        {
          "color": "#D8B9C3",
          "fontFamily": "Poppins-Bold",
          "fontSize": 28,
          "letterSpacing": 1,
          "textAlign": "left",
        }
      }
      testID="selectedwalletarea-header"
    >
      $ 1.000
    </Text>
    <Text
      style={
        {
          "color": "#BDBDBD",
          "fontFamily": "Poppins-Light",
          "fontSize": 15,
          "fontWeight": "400",
          "textAlign": "left",
        }
      }
      testID="selectedwalletarea-label"
    >
      Example Wallet
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 65,
            "justifyContent": "center",
            "opacity": 1,
          },
          false,
          {
            "borderWidth": 0,
            "height": "auto",
          },
        ]
      }
      testID="selectedwalletarea-button"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Poppins-Bold",
              "fontSize": 18,
              "textAlign": "center",
            },
            {
              "color": "white",
              "fontSize": 15,
              "fontWeight": "normal",
            },
          ]
        }
        testID="button-text"
      >
        Change
      </Text>
    </View>
  </View>
</View>
`;
